lambda/app/src/queries/request.ts

```diff
@@ -80,19 +80,28 @@ export const getAllowedRequest = async (session: Session, requestId: number, rol
   return getMyOrTeamRequest(session.user.id, requestId, roles);
 };
 
-export const listIntegrationsForTeam = async (session: Session, teamId: number, options?: { raw: boolean }) => {
+export const listIntegrationsForTeam = async (
+  session: Session,
+  teamId: number,
+  serviceType?: string,
+  options?: { raw: boolean },
+) => {
   if (isAdmin(session)) {
+    const where: any = { teamId, apiServiceAccount: false, archived: false };
+
+    if (serviceType) where.serviceType = serviceType;
     return models.request.findAll({
-      where: { teamId, apiServiceAccount: false, archived: false },
+      where,
       ...options,
     });
   }
 
   const { user } = session;
   const where: any = { apiServiceAccount: false, archived: false };
-
   const teamIdsLiteral = getMyTeamsLiteral(user.id);
 
+  if (serviceType) where.serviceType = serviceType;
```

Comment:
is there a reason this is assigned twice?  naively  it looks like it could just go once  above the isAdmin block?

Reply:
The `where` constant is defined twice for each scenario (admin and non-admin) so we need to update the servicetype for both the scenarios